RELEASE/scripts/autoscend/quests/level_11.ash

```diff
@@ -1300,6 +1304,10 @@ boolean L11_aridDesert()
 	{
 		int need = 100 - get_property("desertExploration").to_int();
 		auto_log_info("Getting some ultrahydrated, I suppose. Desert left: " + need, "blue");
+		if(!get_property("oasisAvailable").to_boolean() && have_effect($effect[Ultrahydrated]) == 0)
+		{
+			autoadv(1, $location[The Arid\, Extra-Dry Desert]);
```

Comment:
```suggestion
			return autoadv(1, $location[The Arid\, Extra-Dry Desert]);
```

Reply:
Autoscend is essentially a really complex nested if structure. When a function returns true, it starts all over again. Anytime we adv, we should ensure it returns immediately or shortly after to restart the loop. Helps ensure proper actions are taken. In this case this adv should give us ultrahydrated as this is the first adv in the desert. No need to go to the oasis. However without the return, we will adv in the oasis.

Reply:
Makes sense.